File: 1309. Decrypt String from Alphabet to Integer Mapping/Solution.java
class Solution {
    public String freqAlphabets(String s) {
        char base = 'a' - 1;
        StringBuilder sb = new StringBuilder();
        for (int i = s.length() - 1; i >= 0;) {
            if (s.charAt(i) == '#') {
                sb.append((char)((s.charAt(i - 2) - '0') * 10 + s.charAt(i - 1) - '0' + base));
                i -= 3;
            } else {
                sb.append((char)(s.charAt(i) - '0' + base));
                i--;
            }
        }
        return sb.reverse().toString();
    }
}

File: 1309. Decrypt String from Alphabet to Integer Mapping/Solution.cpp
#include <algorithm>
#include <string>
using namespace std;

class Solution
{
public:
    string freqAlphabets(string s)
    {
        string res;
        int i = s.size() - 1;
        char base = 'a' - 1;
        while (i >= 0) {
            if (s[i] == '#') {
                res.push_back((s[i - 2] - '0') * 10 + (s[i - 1] - '0') + base);
                i -= 3;
            }
            else {
                res.push_back(s[i] - '0' + base);
                i--;
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};
